lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -557,6 +557,27 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def confirm_ejson_keys_not_prunable
+      raw_secret, st, err = nil
+      with_retries(2) do
+        raw_secret, err, st = kubectl.run("get", "secret",
+          EjsonSecretProvisioner::EJSON_KEYS_SECRET, "-oyaml", output_is_sensitive: true)
+        st.success? || err.include?(KubernetesDeploy::Kubectl::NOT_FOUND_ERROR_TEXT)
+      end
+      if st.success?
+        secret = YAML.safe_load(raw_secret, [Date, Time])
+        return unless secret.dig("metadata", "annotations", KubernetesResource::LAST_APPLIED_ANNOTATION)
+        @logger.error("Secret #{EjsonSecretProvisioner::EJSON_KEYS_SECRET} will be pruned if deploy proceeds")
+
+        raise EjsonPrunableError.exception(
```

Comment:
why `.exception` instead of `.new`? I had to look up what `.exception` does and it seems to have the exact same result.

I'm wondering if the pattern you want is actually `@logger.summary.add_paragraph(this_message)` followed by `raise FatalDeploymentError, "Configuration invalid"` like L358